basesandbox/src/execution/executor.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+
+use crate::ipc::semaphore::{Mode, Semaphore};
+use crate::ipc::Ipc;
+use std::process::Command;
+
+const SOCKET_SRC: &str = "./Foundry_Socket_Boxer";
+const SOCKET_DST: &str = "./Foundry_Socket_Boxee";
+const SEMAPHORE: &str = "/Foundry_Semaphore";
+
+struct Executor {
+    child: std::process::Child,
+}
+
+impl Executor {
+    fn new(path: &str, args: &[&str]) -> Self {
+        let mut command = Command::new(path);
+        for x in args {
+            command.arg(x);
+        }
+        Executor {
+            child: command.spawn().unwrap(),
+        }
+    }
+}
+
+impl Drop for Executor {
+    fn drop(&mut self) {
+        self.child.wait().unwrap();
```

Comment:
Is it okay not to kill the child process and just wait?

Reply:
I think it depends on Foundry's overall module error handling policy. If the parent needs to kill the child explicitly, then it would indicate that something abnormal happened to child, since we expect it to terminate itself by some protocol.

Reply:
I cannot see the protocol which terminates the child process. Isn't it implemented yet?

Reply:
@sgkim126 "it to terminate itself" means child just exiting by returning the `main()`. It happens already in this code.